app/uk/gov/hmrc/lingua/NameCase.scala

```diff
@@ -0,0 +1,141 @@
+      s = s match {
+        case p(a, b, c, d) => a + b + c.capitalize + d
+        case _             => s
+      }
+      s = raw"(\bMacEvicius)".r.replaceAllIn(s, "Macevicius")
+      s = raw"(\bMacHado)".r.replaceAllIn(s, "Machado")
+      s = raw"(\bMacHar)".r.replaceAllIn(s, "Machar")
+      s = raw"(\bMacHin)".r.replaceAllIn(s, "Machin")
+      s = raw"(\bMacHlin)".r.replaceAllIn(s, "Machlin")
+      s = raw"(\bMacIas)".r.replaceAllIn(s, "Macias")
+      s = raw"(\bMacIulis)".r.replaceAllIn(s, "Maciulis")
+      s = raw"(\bMacKie)".r.replaceAllIn(s, "Mackie")
+      s = raw"(\bMacKle)".r.replaceAllIn(s, "Mackle")
+      s = raw"(\bMacKlin)".r.replaceAllIn(s, "Macklin")
+      s = raw"(\bMacQuarie)".r.replaceAllIn(s, "Macquarie")
+      s = raw"(\bMacOmber)".r.replaceAllIn(s, "Macomber")
+      s = raw"(\bMacIn)".r.replaceAllIn(s, "Macin")
+      s = raw"(\bMacKintosh)".r.replaceAllIn(s, "Mackintosh")
+      s = raw"(\bMacKen)".r.replaceAllIn(s, "Macken")
+      s = raw"(\bMacHen)".r.replaceAllIn(s, "Machen")
+      s = raw"(\bMacisaac)".r.replaceAllIn(s, "MacIsaac")
+      s = raw"(\bMacHiel)".r.replaceAllIn(s, "Machiel")
+      s = raw"(\bMacIol)".r.replaceAllIn(s, "Maciol")
+      s = raw"(\bMacKell)".r.replaceAllIn(s, "Mackell")
+      s = raw"(\bMacKlem)".r.replaceAllIn(s, "Macklem")
+      s = raw"(\bMacKrell)".r.replaceAllIn(s, "Mackrell")
+      s = raw"(\bMacLin)".r.replaceAllIn(s, "Maclin")
+      s = raw"(\bMacKey)".r.replaceAllIn(s, "Mackey")
+      s = raw"(\bMacKley)".r.replaceAllIn(s, "Mackley")
+      s = raw"(\bMacHell)".r.replaceAllIn(s, "Machell")
+      s = raw"(\bMacHon)".r.replaceAllIn(s, "Machon")
+    }
+    s = raw"(\bMacmurdo)".r.replaceAllIn(s, "MacMurdo")
+
+    // Arabic
+    //s = raw"(\bAl +\b)".r.replaceAllIn(s, "al ")
+    // Welsh
+    s = raw"(\bAp +\b)".r.replaceAllIn(s, "ap ")
+    s = raw"(\bVon +\b)".r.replaceAllIn(s, "von ")
+    s = raw"(\bVan +\b)".r.replaceAllIn(s, "van ")
+
+    // della and delle Italian
+    s = raw"(\bDella +\b)".r.replaceAllIn(s, "della ")
+    s = raw"(\bDelle +\b)".r.replaceAllIn(s, "delle ")
+
+    // da, de di Italian
+    s = raw"(\bDa +\b)".r.replaceAllIn(s, "da ")
+    s = raw"(\bDe +\b)".r.replaceAllIn(s, "de ")
+    s = raw"(\bDi +\b)".r.replaceAllIn(s, "di ")
+
+    // du French
+    s = raw"(\bDu +\b)".r.replaceAllIn(s, "du ")
+
+    // du Italian
+    s = raw"(\bDel +\b)".r.replaceAllIn(s, "del ")
+
+    // Dutch/French
+    s = raw"(\bDer +\b)".r.replaceAllIn(s, "der ")
+
+    // El - Spanish or el Greek  - keep Spanish version
+    s = raw"(\bEl +\b)".r.replaceAllIn(s, "El ")
+
+    // Italian
+    s = raw"(\bLe +\b)".r.replaceAllIn(s, "le ")
+
+    // French
+    s = raw"(\bLo +\b)".r.replaceAllIn(s, "lo ")
+
+    // la French or La Spanish - keep  Spanish version
+    s = raw"(\bLa +\b)".r.replaceAllIn(s, "La ")
+
+    POST_NOMINAL_INITIALS.foreach(
+      n => s = s.replaceFirst(raw"(?i)\b$n" + " *$", s"$n")
```

Comment:
replaceFirst -> replaceLast ?

Reply:
it doesn't matter, it has "$" at the end of the regex